kb/0002-scripting-overview.md

```diff
@@ -40,6 +40,461 @@ The following list is sorted alphabetically and is based on the latest build of
 * `program\weather\` - weather system definitions and usage.
 * `program\worldMap\` - everything world map related.
 
+## Typing
+
+In order to be compiled, the scripting files should contain good C-like code. This also means that the variables should be strongly typed. Commonly used are base C types: `int`, `float`, but also `bool` and a simple to use `string`. Any custom structs are supported in the form of `object`s. Last, arrays are supported of the above.
+
+### Base Types
+
+There're three base types in scripting language: 
+
+* `int`: a signed 32-bit integer
+
+``` C++
+syntax: 
+	int var_name;
+examples:
+	int GameTime = -2053; // OK
+	int FreeSlots = 75;   // OK
+```
+
+* `float`: a single-precision floating-point number. Because there're no `double`s, specifying `f` at the end is not required but allowed. 
+
+``` C++
+syntax: 
+	float var_name;
+examples:
+	float WindSpeed = 2.0;         // OK
+	float WindDirection = -0.0057; // OK
+	float ShipDirection = 3.14f;   // OK
+```
+
+* `bool`: a logical `true`/`false` operator where `true` is any non-zero value and `false` is zero.
+
+``` C++
+syntax: 
+	bool var_name;
+examples:
+	bool IsSpeaking = true;                  // OK
+    bool IsDrunk = (AlcoholInBlood > 10.0f); // OK
+    bool KilledAtLeastOneEnemy = KillCount;  // OK where KillCount is an int
+```
+
+### Strings
+
+In the scripting code, strings are sequences of characters of arbitrary length. You can concatenate multiple strings together using `+` operator. 
+
+``` C++
+syntax: 
+	string var_name;
+example:
+	string Name = "John";                    // OK
+    string LastName = GetRandomLastName();   // OK, e.g. "Jameson"
+    string FullName = Name + " " + LastName; // OK, "John Jameson"
+```
+
+### Arrays
+
+The scripting code allows fixed-size arrays of any type. This means that, after being initialized, the length of the array cannot be modified. However, unlike in C, the array _may_ be initialized from a variable. In that case, the array size would be that variable's value at the array creation time. 
+
+``` C++
+syntax: 
+    var_type var_name[int number_of_elements];
+example:
+	int Characteristics[10];                // OK
+	string CharacteristicsNames[10];        // OK
+
+	int array_size;
+	array_size = 10;
+	string CharacteristicsNames[array_size]; // Also OK
+```
+
+// TODO(yakvi): Info below is dated 2009. Is this still relevant?
```

Comment:
@mitrokosta Could you help us out here, is it still the case?

Reply:
Most likely yes. According to [this](https://github.com/storm-devs/storm-engine/commit/76e8eaf9cde9ed95dc5b65e34e57a3613d387242) commit's diff compiler's array handling mechanism was not changed.